### ISSUE TITLE
Crontab: Sensitive to spaces and several functionalities are not supported anymore -but documented

### ISSUE THREAD

**Issue by henfri** [smarthomeNG/smarthome#335]
Hello,

`- '0 9  * * = 1'` will not switch
whereas 
`- '0 9 * * = 1'` will

`2019-11-24 16:31:10 ERROR    scheduler    lib.scheduler    Error parsing crontab "0 9  * *": too many values to unpack (expected 4) -- scheduler.py:_crontab:548`

I propose to delete all intermediate spaces between characters if they are more than one before passing to crontab.

Furthermore:

`15 22 * 1,2,3,4,5,6,7` 
`0 10 * 5-6`
are not supported, but documented:

`2019-11-24 16:31:10 ERROR    scheduler    lib.scheduler    Error parsing crontab "15 22 * 1,2,3,4,5,6,7": list index out of range -- scheduler.py:_crontab:548`

`2019-11-24 15:28:22 ERROR scheduler lib.scheduler Error parsing crontab "0 10 * 5-6": invalid literal for int() with base 10: '5-6' -- scheduler.py:_crontab:548`

Regards,
Hendrik

**Comment by bmxp** [comment]
It really can't be seen here: the switch not working contains two spaces between 9 and * and this causes a failure. 
Please check if https://github.com/smarthomeNG/smarthome/commit/666544bcfdb62ca01cfebabc76cb337e25af9ddf#diff-f7a7e9960a35e3a7f3a747b7883f55cf fixes it.

wday should have a range of 0 to 6 where 0=monday and 6=sunday you exceed it to use 7 so this is working fine I think.

So there is a range 5-6 left which is indeed not working. Please try the following code snippet in ``lib.scheduler.py`` around line 583. This is only a quick try for a fix so please test it thouroughly until we can push it to the repo. As this is old code, I am not going to write tests for it, it needs a overhaul anyway.

```
        if wday == '*' and day == '*':
            day_range = self._day_range('0, 1, 2, 3, 4, 5, 6')
        elif wday != '*' and day == '*':
            day_range = self._range(wday,0,6)
            day_range = self._day_range(','.join(day_range))
        elif wday != '*' and day != '*':
            day_range = self._range(wday,0,6)
            day_range = self._day_range(','.join(day_range))
            day_range = day_range + self._range(day, 0o1, mdays)
        else:
            day_range = self._range(day, 0o1, mdays)
```

**Comment by henfri** [comment]
Hello,

I confirm that both fixes work for me.
Tried with all examples from my initial post.

Thank you!

Greetings,
Hendrik

**Comment by bmxp** [comment]
OK, I will push the fix Info develop branch on weekend unless anyone raises hands.

**Comment by bmxp** [comment]
The announced fix 2a51e42fe0ffd0028f999807741fb49e02daf4c4 is pushed so I will close this for now.

**Comment by henfri** [comment]
Danke!
